src/lib/theme.ts

```diff
@@ -0,0 +1,43 @@
+import { browser } from '$app/environment';
+import { writable } from 'svelte/store';
+
+const themes = ['light', 'dark'];
+
+type Theme = 'light' | 'dark';
+
+const isTheme = (theme: string): theme is Theme => {
+  return themes.includes(theme);
+};
+
+const getFromLocalStorage = (): Theme => {
+  if (browser) {
+    const q = window.matchMedia('(prefers-color-scheme: dark)');
+    const localStoreTheme = window.localStorage.getItem('theme');
+
+    if (localStoreTheme && isTheme(localStoreTheme)) {
+      return localStoreTheme;
+    }
+    if (q.matches) {
+      return 'dark';
+    }
+  }
+  return 'light';
+};
+
+const createTheme = () => {
+  const { subscribe, update } = writable<Theme>(getFromLocalStorage());
+
+  return {
+    subscribe,
+    toogle: () => update((v) => (v === 'light' ? 'dark' : 'light'))
+  };
+};
+
+export const theme = createTheme();
+
+theme.subscribe(($theme) => {
+  if (browser) {
+    window.localStorage.setItem('theme', $theme);
+    document.documentElement.setAttribute('data-theme', $theme);
+  }
+});
```

Comment:
There's a little problem, I think if your OS prefers dark, this state shouldn't be saved into localStorage, I mean, shouldn't be preserved because you didn't chose that theme, it's a default from your OS.
I think only when you click to change the theme, that action should preserve it's state in localStorage.
Imagine if your OS chooses dark or light between time of the day (which is pretty usual), if you connect to the website during the night it'll present as dark, but even without clicking on the theme-switch, the next day at the morning the website will default to dark. Why? Because you opened the website when your OS was dark and that state was preserved
I know it's not much critical, but cool to have

Reply:
mmmmm I see what you mean but i think for now we'll leave it like this! it's a really good recommendation though